app/controllers/application_controller.rb

```diff
@@ -7,15 +7,15 @@ def current_user
     @current_user ||= User.find(session[:user_id]) if session[:user_id]
   end
 
-  def require_user
-    render file: "/public/404" unless current_user
-  end
-
   def render_404
     respond_to do |format|
       format.html { render :file => "#{Rails.root}/public/404", :layout => false, :status => :not_found }
       format.xml  { head :not_found }
       format.any  { head :not_found }
     end
   end
+
+  def require_user
+    render_404 unless current_user
```

Comment:
not related to this specific ticket but updated this method to use existing helper method